org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandler.java

```diff
@@ -82,7 +82,7 @@ public class CodeActionHandler {
 	// Multiple Code Actions are computed in parallel through ForkJoinPool's common pool leading to a race condition where
 	// the last Code Action request to finish processing is not the client's latest request. History size must scale with pool size.
 	public static final ResponseStore<Either<ChangeCorrectionProposalCore, CodeActionProposal>> codeActionStore
-			= new ResponseStore<>(Math.max(ForkJoinPool.getCommonPoolParallelism(), 8));
+			= new ResponseStore<>(Math.max(ForkJoinPool.getCommonPoolParallelism(), 1));
```

Comment:
Is this required ? This would regress https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/3180 .

Reply:
Now code actions are synchronized - https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/ecd3b6167a2e00f23bb1b3ae6b6923ce70e338f0/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java#L757